lib/ansible/plugins/inventory/gcp_compute.py

```diff
@@ -0,0 +1,298 @@
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+DOCUMENTATION = '''
+    name: gcp_compute
+    plugin_type: inventory
+    short_description: Google Cloud Compute Engine inventory source
+    requirements:
+        - requests >= 2.18.4
+        - google-auth >= 1.3.0
+    extends_documentation_fragment:
+        - gcp
+        - constructed
+        - inventory_cache
+    description:
+        - Get inventory hosts from Google Cloud Platform GCE.
+        - Uses a <name>.gcp.yaml (or <name>.gcp.yml) YAML configuration file.
+    options:
+        zones:
+          description: A list of regions in which to describe GCE instances.
+        projects:
+          description: A list of projects in which to describe GCE instances.
+        filters:
+          description: A dictionary of filter value pairs. Available filters are listed here
+              U(https://cloud.google.com/compute/docs/reference/rest/v1/instances/list)
+'''
+
+EXAMPLES = '''
+plugin: gcp_compute
+zones: # populate inventory with instances in these regions
+  - us-east1-a
+projects:
+  - gcp-prod-gke-100
+  - gcp-cicd-101
+filters:
+  - machineType = n1-standard-1
+  - scheduling.automaticRestart = true AND machineType = n1-standard-1
+
+scopes:
+  - https://www.googleapis.com/auth/compute
+service_account_file: /tmp/service_account.json
+auth_kind: serviceaccount
+'''
+
+from ansible.errors import AnsibleError, AnsibleParserError
+from ansible.module_utils._text import to_native, to_text
+from ansible.module_utils.six import string_types
+from ansible.module_utils.gcp_utils import GcpSession, navigate_hash, GcpRequestException
+from ansible.plugins.inventory import BaseInventoryPlugin, Constructable, Cacheable, to_safe_group_name
+import json
+
+
+# The mappings give an array of keys to get from the filter name to the value
+# returned by boto3's GCE describe_instances method.
+class GcpMockModule(object):
+    def __init__(self, params):
+        self.params = params
+
+    def fail_json(self, *args, **kwargs):
+        raise AnsibleError(kwargs['msg'])
+
+
+class InventoryModule(BaseInventoryPlugin, Constructable, Cacheable):
+
+    NAME = 'gcp_compute'
+
+    def __init__(self):
+        super(InventoryModule, self).__init__()
+
+        self.group_prefix = 'gcp_'
+
+    def _populate_host(self, item):
+        '''
+            :param item: A GCP instance
+        '''
+        hostname = self._get_hostname(item)
+        self.inventory.add_host(hostname)
+        for key in item:
+            self.inventory.set_variable(hostname, key, item[key])
+        self.inventory.add_child('all', hostname)
+
+    def _validate_file(self, path):
+        '''
+            :param path: the path to the inventory config file
+            :return the contents of the config file
+        '''
+        if super(InventoryModule, self).verify_file(path):
+            if path.endswith('.gcp.yml') or path.endswith('.gcp.yaml'):
```

Comment:
I think for consistency we should add `.gcp_compute.ya?ml` as an accepted suffix, since that's the actual plugin name.

Reply:
Yeah, that seems like a sensible change.